docs/custom_plugin.md

```diff
@@ -0,0 +1,429 @@
+2. `Query.py`: Has a class which inherits from `Query` and obtains the necessary data which is then passed to the requirement(s).
+3. `Module.py`: A file with a class which inherits from `Module` and defines the dynamic arguments and which queries to execute for obtaining necessary data.
+4. `Plugin.py`: Tells `pluggy` how to load the plugin so RepoAuditor can see it.
+
+The files would be organized as:
+
+```sh
+$ tree
+.
+├── README.md
+├── ReadmeCheck
+│   ├── Module.py
+│   ├── Plugin.py
+│   ├── Query.py
+│   ├── Requirement.py
+│   └── __init__.py
+└── pyproject.toml
+```
+
+### Requirement
+
+To define a requirement, we define a class which inherits from `Requirement` and implements three methods:
+
+1. `__init__(self) -> None`: The constructor.
+2. `GetDynamicArgDefinitions(self, argument_separator: str) -> dict[str, TypeDefinitionItemType]`: A method which defines requirement specific dynamic arguments.
+3. `_EvaluateImpl(self, query_data: dict[str, Any], requirement_args: dict[str, Any], ) -> Requirement.EvaluateImplResult`: The logic for verifying the requirement.
+
+The definition for the `Requirement.py` file would be the following. Please see the inline comments for details about specific lines of code.
+
+```python title="ReadmeCheck/Requirement.py"
+"""Contains the Readme object."""
+
+import textwrap
+from pathlib import Path
+from typing import Any
+
+import typer
+from dbrownell_Common.TyperEx import \
+    TypeDefinitionItemType  # type: ignore[import-untyped]
+from dbrownell_Common.Types import override  # type: ignore[import-untyped]
+from RepoAuditor.Requirement import EvaluateResult, ExecutionStyle, Requirement
+
+
+class ReadmeRequirement(Requirement):
+    """Requirement to validate a repository's README."""
+
+    def __init__(self) -> None:
+        super().__init__(
+            name="Readme",
+            description="Validates if the repository has a README.",
+            style=ExecutionStyle.Parallel,
+
+            # This is the template for the instructions to resolve
+            # a failure caused by this requirement.
+            resolution_template=textwrap.dedent(
+                """\
+                Add a README file to the repository.
+                """, ),
+
+            # This is the template for the rationale behind why this
+            # requirement is good to have.
+            rationale_template=textwrap.dedent(
+                """\
+                A README file helps users understand the project.
+                """, ),
+
+            # Flag which decides if an explicit include from the
+            # command line is requiremd to enable this requirement.
+            requires_explicit_include=False,
+        )
+
+        # The dynamic_arg_name is used in defining the dynamic
+        # argument `ReadmeCheck-no-Readme`.
+        self.dynamic_arg_name: str = "no"
+
+        # Flag deciding if the requirement is enabled by default.
```

Comment:
Nit: This comment might need some additional information, as it has been such a source of conversation for us. I suggest:

- Moving this above `dynamic_arg_name`, as the meaning of `dynamic_arg_name` is dependent upon this value.

- Provide a counter-example of why a person might set this to False (e.g. to ensure `README.md` is NOT present).